src/spoke/Spoke.sol

```diff
@@ -185,20 +185,20 @@ contract Spoke is Auth, Recoverable, ReentrancyProtection, ISpoke, IUpdateContra
         require(decimals <= MAX_DECIMALS, TooManyDecimals());
         require(isPoolActive(poolId), InvalidPool());
 
-        Pool storage pool = pools[poolId];
-        require(address(pool.shareClasses[scId].shareToken) == address(0), ShareClassAlreadyRegistered());
+        require(address(pools[poolId].shareClasses[scId].shareToken) == address(0), ShareClassAlreadyRegistered());
 
         // Hook can be address zero if the share token is fully permissionless and has no custom logic
         require(hook == address(0) || _isValidHook(hook), InvalidHook());
 
         IShareToken shareToken_ = tokenFactory.newToken(name, symbol, decimals, salt);
 
-        if (hook != address(0)) {
-            shareToken_.file("hook", hook);
-        }
-
-        pool.shareClasses[scId].shareToken = shareToken_;
+        if (hook != address(0)) shareToken_.file("hook", hook);
+        linkToken(poolId, scId, shareToken_);
+    }
 
+    /// @inheritdoc ISpoke
+    function linkToken(PoolId poolId, ShareClassId scId, IShareToken shareToken_) public auth {
+        pools[poolId].shareClasses[scId].shareToken = shareToken_;
```

Comment:
QUESTION

Why did you remove the requires? That here would allow to override share tokens essentially still. No blocker just wanted to highlight that this is a really invasive method now.

Reply:
I'd also prefer moving the check of L188 here as this method is `public auth`. Curious about your train of thought.

Reply:
The requires are still in `addShareClass` rather than `linkToken`